lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -129,11 +129,7 @@ defmodule Mix.Tasks.Xref do
   ## Modes
 
   defp warnings() do
-    if unreachable(&print_warnings/2) == [] do
-      :ok
-    else
-      :error
-    end
+    List.flatten(unreachable(&warnings/2))
```

Comment:
Do we really need `List.flatten`? `List.flatten` is almost always a smell. It means we were supposed to flat map or concat at some point before but we did not.